client/src/app/services/http.service.ts

```diff
@@ -0,0 +1,26 @@
+import { HttpClient, HttpErrorResponse, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Course } from '../models/course.model';
+import { CourseDetail } from '../models/lesson.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HttpService {
+  public error: HttpErrorResponse;
+
+  constructor(private http: HttpClient) { }
+
+  public getAllCourses(paramsObj: { limit: number, offset: number }): Observable<{ courses: Course[], allCoursesLength: string }> {
+    return this.http.get<{ courses: Course[], allCoursesLength: string }>('http://localhost:3000/', {
```

Comment:
You can create a variable for the base URL and use it, then it will be easy to change if needed
 Something like this `const BASE_URL = 'http://localhost:3000/`;